DOC: add SPSS comparison guide structure

Jacob-Lazar opened the pull request:
PR adds a comparison guide for SPSS users transitioning to pandas, addressing #60727. 

## Changes
- Creates new `comparison_with_spss.rst` in getting_started/comparison/
- Documents basic data structure mappings:
  - SPSS data files → pandas DataFrames
  - SPSS variables → pandas Series
  - SPSS case numbers → pandas Index
- Sets up structure for common operations comparison

## Implementation
- Following existing comparison guide formats (R, SQL, etc.)
- Using reStructuredText format
- Including executable code examples
- Cross-referencing pandas documentation

- [ x] closes #60727

Commit:
DOC: add SPSS comparison guide structure

- Create SPSS comparison documentation
- Add header and introduction sections
- Terminology translation table
- Create template for common operations comparison

Part of #60727

afeld reviewed (changes requested):
Thanks so much for hopping on this so fast! Please do the following:

- [ ] Address the various build failures - seems they are minor, but legitimate
- [ ] Add the file to [`doc/source/getting_started/comparison/index.rst`](https://github.com/pandas-dev/pandas/blob/main/doc/source/getting_started/comparison/index.rst)

I don't have experience with SPSS so unfortunately can't give meaningful feedback on the content. I teach students who have experience with both, so I'll try and get a handful of them to take a look.

Commit:
DOC: edit SPSS comparison guide to documentation

- Added file to doc/source/getting_started/comparison/index.rst toctree
- Fixed formatting and whitespace issues to meet documentation standards

Commit:
DOC: edit minor whitespaces in SPSS comparison guide

mroeschke commented on doc/source/getting_started/comparison/comparison_with_spss.rst:
```diff
@@ -0,0 +1,229 @@
+    groupby, split file
+    ``NaN``, system-missing
+
+``DataFrame``
```
For these double ticked words, could you use ``` :class:`Foo` ``` instead?

Jacob-Lazar replied:
Sure

mroeschke commented on doc/source/getting_started/comparison/comparison_with_spss.rst:
```diff
@@ -0,0 +1,229 @@
+
+The pandas equivalent would use :func:`read_csv`:
+
+.. ipython:: python
```
Could you make these all `.. code-block:: python` instead?

Jacob-Lazar replied:
Sure

Jacob-Lazar replied:
I've edited the code

Commit:
DOC: standardize class references in SPSS guide

Commit:
DOC: Fix RST section underline lengths in SPSS comparison

Jacob-Lazar committed:
Merge branch 'main' into main

mroeschke commented:
Thanks @Jacob-Lazar
